api/v1beta1/gcpcluster_types.go

```diff
@@ -54,6 +54,14 @@ type GCPClusterSpec struct {
 	// +optional
 	AdditionalLabels Labels `json:"additionalLabels,omitempty"`
 
+	// ResourceManagerTags is an optional set of tags to apply to GCP resources managed
+	// by the GCP provider. GCP supports a maximum of 50 tags per resource.
+	// +kubebuilder:validation:MaxItems=50
+	// +listType=map
+	// +listMapKey=key
+	// +optional
+	ResourceManagerTags []ResourceManagerTag `json:"resourceManagerTags,omitempty"`
```

Comment:
I see `ResourceManagerTags` on the GCPMachine being used. However, i can't see where the `ResourceManagerTags` on GCPCluster/GCPManagedCluster are being used?

There is a pattern in other providers for tags where there is precedence. Tags can be supplied on the infra cluster (i.e. GCPCluster or GCPManagedCluster for GCP) or on the infra machine (i.e. GCPMachine). If they are both supplied then the tags on GCPMachine are used.